Kernel/LPO.cpp

```diff
@@ -196,30 +196,26 @@ Ordering::Result LPO::lpo(TermList tl1, TermList tl2) const
 {
   CALL("LPO::lpo(TermList, TermList)");
 
+  if(tl1==tl2) {
+    return EQUAL;
+  }
   if(tl1.isOrdinaryVar()) {
     return INCOMPARABLE;
   }
   ASS(tl1.isTerm());
-  return lpo(tl1.term(), tl2);
-}
-
-// used when the first term is not a variable
-Ordering::Result LPO::lpo(Term* t1, TermList tl2) const
-{
-  CALL("LPO::lpo(Term*, TermList)");
-
+  Term* t1 = tl1.term();
   ASS(t1->shared());
 
   if(tl2.isOrdinaryVar()) {
     return t1->containsSubterm(tl2) ? GREATER : INCOMPARABLE;
   }
-  
+
   ASS(tl2.isTerm());
   Term* t2=tl2.term();
 
   switch (compareFunctionPrecedences(t1->functor(), t2->functor())) {
   case EQUAL:
-    lexMAE(t1, t2, t1->args(), t2->args(), t1->arity());
```

Comment:
This is horrifying! We should probably use some compiler machinery for warning about unused return values, but that's a separate PR.

Reply:
Agreed, it should be quite trivial to catch these.